crud.c

```diff
@@ -0,0 +1,302 @@
+    User User;
+    if (!WriteFilePointer)
+    {
+        perror("Error writing to file");
+        return;
+    }
+    User.Id = GetId();
+    if (UserPresent(User.Id))
+    {
+        printf("----User ID already exists----\n");
+        fclose(WriteFilePointer);
+        return;
+    }
+    GetName(User.Name, sizeof(User.Name));
+    User.Age = GetAge();
+    WriteUser(WriteFilePointer, &User);
+    printf("\n----Data entered successfully-----\n");
+    fclose(WriteFilePointer);
+}
+
+void Read()
+{
+    FILE *ReadFilePointer = fopen("users.txt", "r");
+    char BufferString[100];
+    if (ReadFilePointer == NULL)
+    {
+        perror("----Error opening file----");
+        exit(EXIT_FAILURE);
+    }
+    while (fgets(BufferString, sizeof(BufferString), ReadFilePointer))
+    {
+        printf("%s", BufferString);
+    }
+    fclose(ReadFilePointer);
+}
+
+void Update()
+{
+    FILE *ReadFilePointer = fopen("users.txt", "r");
+    FILE *TempFilePointer = fopen("temp.txt", "w");
+    User User;
+    int TargetId;
+    char BufferString[100];
+    if (!TempFilePointer || !ReadFilePointer)
+    {
+        printf("Error opening file");
+        fclose(TempFilePointer);
+        fclose(ReadFilePointer);
+        return;
+    }
+    TargetId = GetId();
+    if (UserPresent(TargetId))
+    {
+        while (fgets(BufferString, sizeof(BufferString), ReadFilePointer))
+        {
+            if (ReadUser(BufferString, &User) == 3)
+            {
+                if (User.Id == TargetId)
+                {
+                    GetName(User.Name, sizeof(User.Name));
+                    User.Age = GetAge();
+                }
+                WriteUser(TempFilePointer, &User);
+            }
+        }
+    }
+    else
+    {
+        printf("User not found\n");
+        fclose(TempFilePointer);
+        fclose(ReadFilePointer);
+        remove("temp.txt");
+        return;
+    }
+
+    fclose(TempFilePointer);
+    fclose(ReadFilePointer);
+    remove("users.txt");
+    rename("temp.txt", "users.txt");
+    printf(" ----User updated successfully----\n");
+}
+
+void Delete()
+{
+    FILE *ReadFilePointer = fopen("users.txt", "r");
+    FILE *TempFilePointer = fopen("temp.txt", "w");
+    User User;
+    int TargetId;
+    char BufferString[100];
+
+    if (!TempFilePointer || !ReadFilePointer)
+    {
+        printf("----Error opening file----");
+        return;
+    }
+    TargetId = GetId();
+    if (UserPresent(TargetId))
+    {
+        while (fgets(BufferString, 100, ReadFilePointer))
+        {
+            if (ReadUser(BufferString, &User) == 3)
+            {
+                if (User.Id != TargetId)
+                {
+                    WriteUser(TempFilePointer, &User);
+                }
+            }
+        }
+    }
+    else
+    {
+        printf("----User ID not present, Please Enter a valid ID\n----");
+        fclose(ReadFilePointer);
+        fclose(TempFilePointer);
+        remove("temp.txt");
+        return;
```

Comment:
Is it required to return from the middle of function, can it be handled in some other way